lib/BuildSystem/LaneBasedExecutionQueue.cpp

```diff
@@ -393,8 +423,27 @@ class LaneBasedExecutionQueue : public BuildExecutionQueue {
     posix_spawn_file_actions_init(&fileActions);
 
     // Open /dev/null as stdin.
-    posix_spawn_file_actions_addopen(
-        &fileActions, 0, "/dev/null", O_RDONLY, 0);
+//    posix_spawn_file_actions_addopen(
+//        &fileActions, 0, "/dev/null", O_RDONLY, 0);
+
+
+    // Create a pipe for the process to (potentially) release the lane while
+    // still running.
+    int releasePipe[2]{ -1, -1 };
+    if (basic::sys::pipe(releasePipe) < 0) {
+      getDelegate().commandProcessHadError(
+          context.job.getForCommand(), handle,
+          Twine("unable to open lane release pipe (") + strerror(errno) + ")");
+      getDelegate().commandProcessFinished(context.job.getForCommand(),
+                                           handle,
+                                           CommandExtendedResult::makeFailed());
+      completionFn.unwrapIn([](CommandCompletionFn fn){
+        fn(CommandResult::Failed);
+      });
+      return;
+    }
+    posix_spawn_file_actions_addinherit_np(&fileActions, releasePipe[1]);
```

Comment:
Heh, and now I know why the others are using _adddup2.  Linux doesn't support addinherit_np.

Reply:
np stands for non portable. :)

Reply:
Ohhhh.  🤯